components/skus/renderer/skus_js_handler.cc

```diff
@@ -194,9 +194,16 @@ void SkusJSHandler::OnFetchOrderCredentials(
                                  v8::MicrotasksScope::kDoNotRunMicrotasks);
 
   v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
+
+  if (!response.empty()) {
```

Comment:
This is a bit hard to follow at first, I think maybe something like this would be better
```
v8::Local<v8::String> result = v8::String::NewFromUtf8(isolate, response.c_str()).ToLocalChecked();
if (response.empty()) {
  std::ignore = resolver->Resolve(context, result);
} else {
  std::ignore = resolver->Reject(context, result);
}

Reply:
also this needs a test